src/inspector/models/abstractinspectormodel.cpp

```diff
@@ -235,7 +235,7 @@ Ms::Sid AbstractInspectorModel::styleIdByPropertyId(const Ms::Pid pid) const
 
 void AbstractInspectorModel::updateStyleValue(const Ms::Sid& sid, const QVariant& newValue)
 {
-    if (style()) {
+    if (style() && style()->styleValue(sid) != newValue) {
         beginCommand();
```

Comment:
this was causing some truly bizarre and random behaviour - basically the inspector was triggering changes to the undo stack even if nothing had changed, just by clicking on empty parts of the score